cmd/ocm-backplane/managedJob/createManagedJob.go

```diff
@@ -335,3 +352,53 @@ func getJobStatus(client BackplaneApi.ClientInterface, job *BackplaneApi.Job) (B
 
 	return *formatJobResp.JSON200.JobStatus.Status, nil
 }
+
+var listDynaKube = func(cl client.Client,ctx context.Context, u client.ObjectList, opts ...client.ListOption) error{
+	return cl.List(ctx, u, opts...)
+}
+
+func getDynatraceInstance() (instance string, err error) {
+	bp, err := config.GetBackplaneConfiguration()
+	if err != nil {
+		return
+	}
+	cfg, err := login.GetRestConfig(bp, options.clusterID)
+	if err != nil {
+		return
+	}
+	cl, err := client.New(cfg, client.Options{})
+
+	if err != nil {
+		return
+	}
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "dynatrace.com",
+		Kind:    "DynaKube",
+		Version: "v1alpha1",
+	})
+
+	err = listDynaKube(cl,context.Background(), u, client.InNamespace("dynatrace"))
```

Comment:
Managed scripts execute in all types of clusters ( `osd`, `MC`, `SC` , `HCP` etc) .. but `dynatrace` NS only available on MC clusters 
This may not be applicable for other type of clusters

Reply:
I tested this on an HCP as well and it still worked fine. If the function does not find a "dynatrace" namespace, it will just return an empty list, and skip over the `EachListItem` call on line 386.

Is there a better way to know if we're not on an MC so we can skip the call altogether?

Reply:
The `dynatarace` NS may be available on `osd`/ `rosa` clusters as well ( via CU deployment) , the best way to identify the MC cluster is through the labels. 
As I explained in the [Slack thread](https://redhat-internal.slack.com/archives/C016S65RNG5/p1714425548781719?thread_ts=1714155462.938349&cid=C016S65RNG5), this feature perfectly matches for the managed script rather than printing DT url while creating every managed job.

Reply:
Yeah, I agree it might be better to use cluster labels to check if the cluster is a MC **to avoid unnecessary API calls** for clusters where dynatrace is not applicable. 

So, we probably can update the code to include a check for the cluster type using labels before trying to list the dynatrace namespace?

Reply:
+1 to limit this option for MC only

Reply:
I'm not opposed to the idea, but I was waiting for an answer to my above question. To share it again:
> Is there a better way to know if we're not on an MC so we can skip the call altogether?

Reply:
I asked the OCM folks - it looks like we can use the endpoint `/api/clusters_mgmt/v1/clusters/<internal_cluster_id>/external_configuration/labels` and check for a label with a key of `ext-hypershift.openshift.io/cluster-type` and a value of `management-cluster`.

@tokalevasant we can work on making this update together.

Reply:
With the addition of the new function `isManagementCluster`, I believe we can resolve this thread.

Reply:
@AlexVulaj - The updated functions' logic looks good to me.  I think we can unhold for now. BTW, have you tested to create a managed-scripts testjob against the MC to see if it works as expected?

Reply:
@xiaoyu74 It was a bit ago, but yes I tested these changes and all looked good to me.